File: npm/tests/Core_digitalIO/digitalRead.js
const linuxduino = require('linuxduino');

// Test digitalRead
(async() => {
  // Wait for wasm file
  await linuxduino.ready;

  var ledPin = 17; // Embedded board GPIO Number
  var input = 0;
  linuxduino.pinMode(ledPin, linuxduino.INPUT);
  while(1) {
    input = linuxduino.digitalRead(ledPin);
    console.log(input);
    linuxduino.delay(500);
  }
  return 0;
})();

File: tests/Core_digitalIO/digitalRead.cpp
#include "Linuxduino.h"

// Test digitalRead
int main(void) {
    int ledPin = 17; // Embedded board GPIO Number
    int input = 0;
    pinMode(ledPin, INPUT);
    while(1) {
        input = digitalRead(ledPin);
        printf("%d\n", input);
        delay(500);
    }
    return 0;
}